sentry/src/db/campaign.rs

```diff
@@ -49,6 +45,51 @@ pub async fn fetch_campaign(pool: DbPool, campaign: &Campaign) -> Result<Campaig
     Ok(Campaign::from(&row))
 }
 
+pub async fn get_campaigns_for_channel(pool: &DbPool, campaign: &Campaign) -> Result<Vec<Campaign>, PoolError> {
+    let client = pool.get().await?;
+    let statement = client.prepare("SELECT id, channel, creator, budget, validators, title, pricing_bounds, event_submission, ad_units, targeting_rules, created, active_from, active_to FROM campaigns WHERE channel_id = $1").await?;
+
+    let row = client.query(&statement, &[&campaign.channel.id()]).await?;
+
+    let campaigns = row.into_iter().map(|c| Campaign::from(&c)).collect();
+    Ok(campaigns)
+}
+
+pub async fn campaign_exists(pool: &DbPool, campaign: &Campaign) -> Result<bool, PoolError> {
+    let client = pool.get().await?;
+    let statement = client
+        .prepare("SELECT EXISTS(SELECT 1 FROM campaigns WHERE id = $1)")
+        .await?;
+
+    let row = client.execute(&statement, &[&campaign.id]).await?;
+
+    let exists = row == 1;
+    Ok(exists)
+}
+
+pub async fn update_campaign_in_db(pool: &DbPool, campaign: &Campaign) -> Result<bool, PoolError> {
+    let client = pool.get().await?;
+    let statement = client
+        .prepare("UPDATE campaigns SET budget = $1, validators = $2, title = $3, pricing_bounds = $4, event_submission = $5, ad_units = $6, targeting_rules = $7 WHERE id = $8")
+        .await?;
+
+    let row = client
+        .execute(&statement, &[
+            &campaign.budget,
+            &campaign.validators,
+            &campaign.title,
+            &campaign.pricing_bounds,
+            &campaign.event_submission,
+            &campaign.ad_units,
```

Comment:
Please make sure to test this function since `ad_units` might need to be wrapped in a `Json(campaign.ad_units)`